compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/WindowLocationTracker.desktop.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.window
+
+import java.awt.GraphicsConfiguration
+import java.awt.GraphicsEnvironment
+import java.awt.Point
+import java.awt.Toolkit
+import java.awt.Window
+import java.awt.event.WindowEvent
+import java.awt.event.WindowFocusListener
+
+/**
+ * Track position of all opened windows and provide an appropriate location for new created windows.
+ *
+ * Needed to place windows in cascade, and on the same screen.
+ *
+ * Singleton because we have only the single platform.
+ * We basically override the standard behaviour of the window manager.
+ */
+internal object WindowLocationTracker {
+    private val cascadeOffset = Point(48, 48)
+
+    private var lastFocusedWindows = mutableSetOf<Window>()
+
+    private val focusListener = object : WindowFocusListener {
+        override fun windowGainedFocus(e: WindowEvent) {
+            // put window on the top of the set
```

Comment:
What's the benefit of moving it to the front of the linked set? 

If it's important that it should be `LinkedHashSet` then maybe it should be used explicitly instead of `mutableSetOf`? Since mutableSetOf returns `MutableSet` (so not guaranteed to be LinkedHashSet)

Reply:
I see why it needs to be ordered:

```
val lastWindow = lastFocusedWindows.lastOrNull()
```

So the first part of the question is solved

Reply:
> If it's important that it should be LinkedHashSet then maybe it should be used explicitly instead of mutableSetOf?

 `mutableSetOf` preserves the order by its specification:
`Elements of the set are iterated in the order they were specified.`

so we can safely use it